packages/component-base/src/iron-list-core.js

```diff
@@ -627,6 +627,7 @@ export const ironList = {
       forceUpdate ||
       this._scrollHeight === 0 ||
       this._scrollPosition >= estScrollHeight - this._physicalSize ||
+      estScrollHeight >= this._maxPages * this._viewportHeight ||
```

Comment:
See [this note](https://github.com/vaadin/web-components/blob/761d2e47235001fbcb6ed3dafe6a4ea47b4d794e/packages/component-base/src/iron-list-core.js#L18)

Reply:
Yep. Do you think the solution makes sense in general? I can refactor it into the adapter.

Reply:
The issue with the `this.grid` approach is that I think `this.grid` in the adapter does not seem to work for a default grid anymore. I am not sure if this is customized for another reason.

Reply:
Moved the logic to the adapter. It is a little ugly as the overridden iron list function is possibly called twice. However, it should not affect the performance.

Reply:
Regarding `this.grid`, I think [this block](https://github.com/vaadin/web-components/blob/main/packages/component-base/src/virtualizer-iron-list-adapter.js#L375-L377) can be removed if I am not missing something.